src/lua/popen.c

```diff
@@ -1632,7 +1631,9 @@ lbox_popen_kill(struct lua_State *L)
 /**
  * Wait until a child process get exited or signaled.
  *
- * @param handle  a handle of process to wait
+ * @param handle        a handle of process to wait
+ * @param opts.timeout  time quota in seconds
+ *                      (default: 100 years)
```

Comment:
I'm not sure we should wrap the timeout in a table - we don't usually do that, for example see [`<cond_object:wait([timeout])`](https://www.tarantool.io/en/doc/latest/reference/reference_lua/fiber/#lua-function.cond_object.wait). Are there any other options expected? Anyway, even if there are, we can extend a timeout argument to a table later, when we need it.

Also, I don't think it's worth saying that default is 100 years. The default value is an infinite tiemeout; 100 years is an implementation detail.

Reply:
I don't expect other options, but who knows. My motivation to add the table with options:

* All other popen methods are implemented this way. Moreover, `<popen handle>:read()` and `<popen handle>:write()` accept `timeout` this way.
* Sure, it is possible to extend the API to accept `number|table`, but such polymorphic APIs looks ugly and are hard to present in the documentation ([example](https://www.tarantool.io/en/doc/latest/reference/reference_lua/socket/#lua-function.socket_object.read)).
* `:wait({timeout = 1})` is easier to undertand in the code comparing to `:wait(1)`.

In general, I think that when we add an option, it is better to add a table with options, to don't end up with a polymorphic API in a future.

OTOH, fiber cond and socket has `:wait()` methods, which accept timeout directly as `:wait(<timeout>)`. But if `<socket handle>:wait()` would accept a table, maybe there was no need in separate `<socket handle>:readable()` and `<socket handle>:writeable()` methods. `:wait({timeout = 1, event = socket.opts.R|W|RW})` looks better, as for me.

My personal opinion is that `{timeout = <...>}` has more advantages than just `timeout`.

(Regarding 100 years -- I agree, will replace.)

Reply:
If it's consistent with `popen.read` and `popen.write`, then it should be fine.

Reply:
Left as `{timeout = <...>}`.